src/controller/department.controller.ts

```diff
@@ -0,0 +1,132 @@
+import express, { NextFunction } from "express";
+import EmployeeService from "../service/employee.service";
+import authenticate from "../middleware/authenticate.middleware";
+import Authorize from "../middleware/authorize.middleware";
+import CreateDepartmentDto from "../dto/create-department.dto";
+import { plainToInstance } from "class-transformer";
+import { validate } from "class-validator";
+import ValidationException from "../exception/validation-exception";
+import DepartmentService from "../service/department.service";
+import PutDepartmenDto from "../dto/put-department.dto";
+import { Role } from "../utils/role.enum";
+import logger from "../logger/logger";
+import { MetadataWithSuchNameAlreadyExistsError } from "typeorm";
+import HttpException from "../exception/http.exception";
+
+class DepartmentController{
+    public router: express.Router;
+
+    constructor(private departmentService: DepartmentService){
+        this.router = express.Router();
+
+        this.router.get("/",authenticate,this.getAllDepartments);
+        this.router.get("/:id",authenticate,this.getDepartmentById);
+        this.router.put("/:id",authenticate,this.putDepartmentById);
+        this.router.post("/",authenticate,Authorize([Role.HR]),this.postDepartment);
+        this.router.delete("/:id",authenticate,Authorize([Role.HR]),this.deleteDepartment);
+    }
+
+
```

Comment:
dto validation can be implemented as a middleware since dto code is being reused everywhere